### PR TITLE
fix bug of sys_rt_sigaction, sys_poll, add sys_clock_nanosleep, sys_g…

### PR DESCRIPTION
**Pull request by lhw2002426**
…etsockopt

### CODE REVIEW

File: api/ruxos_posix_api/src/imp/net.rs [hunk 1]
```diff
@@ -21,6 +21,26 @@ use ruxnet::{TcpSocket, UdpSocket};
 use crate::ctypes;
 use crate::utils::char_ptr_to_str;
 
+const SOL_SOCKET: i32 = 1;
```
Comment (ken4647) [inline api/ruxos_posix_api/src/imp/net.rs:24]: Remove these definations and use `ctyes::SO_BROADCAST` to make sure they are consistent, if there is no special reason.

File: api/ruxos_posix_api/src/imp/net.rs [hunk 2]
```diff
@@ -601,6 +621,67 @@ pub unsafe fn sys_getsockname(
...
+                                0
+                            }
+                        }
+                    },
+                    SO_TYPE => match &*socket {
+                        Socket::Udp(_) => ctypes::SOCK_DGRAM,
+                        Socket::Tcp(_) => ctypes::SOCK_STREAM,
+                    },
+                    SO_RCVLOWAT | SO_SNDLOWAT => 1,
+                    SO_BROADCAST => 1,
```
Comment (ken4647) [inline api/ruxos_posix_api/src/imp/net.rs:667]: Is there any reason not to merge `SO_RCVLOWAT | SO_SNDLOWAT => 1,` and `SO_BROADCAST => 1,`.

File: api/ruxos_posix_api/src/imp/time.rs [hunk 3]
```diff
@@ -74,6 +74,50 @@ pub unsafe fn sys_clock_settime(_clk: ctypes::clockid_t, ts: *const ctypes::time
...
+/// TODO: deal with flags
+pub unsafe fn sys_clock_nanosleep(
+    which_clock: ctypes::clockid_t,
+    flags: c_int,
+    req: *const ctypes::timespec,
+    rem: *mut ctypes::timespec,
+) -> c_int {
+    syscall_body!(sys_clock_nanosleep, {
+        unsafe {
+            if req.is_null() || (*req).tv_nsec < 0 || (*req).tv_nsec > 999999999 {
```
Comment (ken4647) [inline api/ruxos_posix_api/src/imp/time.rs:89]: It would be better to replace `999999999` with `const NANO_PER_SECOND = 1000000000` .

Comment (ken4647) [reply api/ruxos_posix_api/src/imp/time.rs:89]: and leave a comment here: )

File: api/ruxos_posix_api/src/imp/net.rs (hunk 2 above)
Comment (lhw2002426) [reply api/ruxos_posix_api/src/imp/net.rs:667]: I thinked it may not impl in same time in the future, but merge them is also ok.

**Review by ken4647 — APPROVED** [review]

**Pull request merged**